docs/2019-03-28-All-Path-Reachability-Proofs.md

```diff
@@ -0,0 +1,346 @@
+Proving All-Path Reachability Claims
+====================================
+
+This document details All-Path Reachability without solving the
+most-general-unifier (MGU) problem.
+MGU will be detailed in a separate document.
+
+_Prepared by Traian Șerbănuță, Virgil Șerbănuță, Xiaohong Chen._
+
+Background
+----------
+
+Recommended reading: [proving One-Path Reachability
+claims](2018-11-08-One-Path-Reachability-Proofs.md).
+
+Similarly to the One-Path Reachability document, we assume all pattern variables
+used in this document to be _extended function-like patterns_, that is patterns
+which can be written as `t ∧ p` where the interpretation of `t` contains at most
+one element and `p` is a predicate.
+
+_Extended constructor patterns_ will be those extended function-like patterns
+for which `t` is a functional term, composed out of constructor-like symbols
+and variables.
+
+
+__Note:__
```

Comment:
I would be inclined to call the two immediately-following statements "Lemma 1a" and "Lemma 1b" and refer back to them as needed below.